purism-ectool: init at 0.3.5

avieth opened the pull request:
###### Motivation for this change

Adds the os-specific/linux Purism EC tool

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

avieth committed:
maintainers: add avieth

avieth committed:
purism-ectool: init at 0.3.5

Useful tool for users of NixOS on Purism laptops.

SuperSandro2000 commented on pkgs/os-specific/linux/purism/ectool.nix:
```diff
@@ -0,0 +1,28 @@
+{ lib, fetchgit, rustPlatform, udev, pkg-config }:
+let
+
+  gitsrc = fetchgit {
+      url = "https://source.puri.sm/firmware/librem-ec.git";
+      rev = "05d9990d7500e1cf3d09844407f77cb64c049a12";
+      sha256 = "145x8v5zmc9vp9hci0hf3lczyi1bf8hj4bawyy5vgy8bpvdsjgk7";
+    };
```
```suggestion
  gitsrc = fetchgit {
    url = "https://source.puri.sm/firmware/librem-ec.git";
    rev = "05d9990d7500e1cf3d09844407f77cb64c049a12";
    sha256 = "145x8v5zmc9vp9hci0hf3lczyi1bf8hj4bawyy5vgy8bpvdsjgk7";
  };
```
also this should use fetchFromGitLab.

SuperSandro2000 commented on pkgs/os-specific/linux/purism/ectool.nix:
```diff
@@ -0,0 +1,28 @@
+
+  src = gitsrc + /tool;
+
+  cargoSha256 = "sha256:16ahi0qpwfk7vc2wymishg19rbyadv4sk1hp0qzsjj0m3s9ws3yh";
```
```suggestion
  cargoSha256 = "16ahi0qpwfk7vc2wymishg19rbyadv4sk1hp0qzsjj0m3s9ws3yh";
```

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -326,6 +326,8 @@ with pkgs;
 
   deviceTree = callPackage ../os-specific/linux/device-tree {};
 
+  purism-ec-tool = callPackage ../os-specific/linux/purism/ectool.nix {};
```
```suggestion
  purism-ec-tool = callPackage ../os-specific/linux/purism/ectool {};
```

avieth committed:
purism-ectool: style fixes

r-rmcgibbo commented:
Result of `nixpkgs-review pr 134999` at ea4302c8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>purism-ectool</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/os-specific/linux/purism/ectool/default.nix:27:5:
```
   |
27 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

avieth committed:
prism-ectool: license is gpl3Plus

Section 14 of the license at the source repository says

Each version is given a distinguishing version number.  If the
Program specifies that a certain numbered version of the GNU General
Public License "or any later version" applies to it, you have the
option of following the terms and conditions either of that numbered
version or of any later version published by the Free Software
Foundation.  If the Program does not specify a version number of the
GNU General Public License, you may choose any version ever published
by the Free Software Foundation.

doronbehar commented:
@avieth do you wish this to get merged? There are merge conflicts now but it seems good otherwise.

avieth commented:
> @avieth do you wish this to get merged? There are merge conflicts now but it seems good otherwise.

Turns out Purism grifts faulty hardware, therefore I don't want to be a maintainer on this software. Perhaps we should abandon this.
